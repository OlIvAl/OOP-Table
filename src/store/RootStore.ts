import { FinanceActionsStore } from './Actions/FinanceActionsStore';
import { IFinanceActionModel } from './Actions/interfaces';
import { IStoreWithCurrentModel } from './Abstract/interfaces';

export class RootStore {
  financeActionsStore: IStoreWithCurrentModel<IFinanceActionModel>;

  constructor() {
    this.financeActionsStore = new FinanceActionsStore(this);
  }
}