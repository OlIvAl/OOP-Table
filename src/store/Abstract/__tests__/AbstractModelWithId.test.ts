import TestStoreWithModelsWithId from '../Test/TestStoreWithModelsWithId';
import { IStore } from '../interfaces';
import { ITestModelWithId } from '../Test/interfaces';
import TestModelWithId from '../Test/TestModelWithId';
import { RootStore } from '../../RootStore';

function getRandomInt() {
  return Math.floor(Math.random());
}

let rootStore = new RootStore();
let store: IStore<ITestModelWithId>;
let model: ITestModelWithId;
let id: number;

beforeEach(() => {
  id = getRandomInt();
  store = new TestStoreWithModelsWithId(rootStore);
  model = new TestModelWithId(store, id);
});

describe('AbstractModel', () => {
  describe('constructor', () => {
    it('init model with valid store', () => {
      expect(model).toMatchSnapshot();
      expect(model.store).toEqual(store);
    });
    it('init model with valid id', () => {
      expect(model.id).toEqual(id);
    });
  });
});
