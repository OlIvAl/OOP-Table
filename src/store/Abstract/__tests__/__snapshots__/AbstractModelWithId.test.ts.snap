// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbstractModel constructor init model with valid store 1`] = `
TestModelWithId {
  "id": 0,
  "store": TestStoreWithModelsWithId {
    "rootStore": RootStore {
      "financeActionsStore": FinanceActionsStore {
        "rootStore": [Circular],
      },
    },
  },
}
`;
